service/src/main/java/bio/terra/axonserver/service/cromwellworkflow/CromwellWorkflowService.java

```diff
@@ -245,34 +259,39 @@ public CromwellApiWorkflowIdAndStatus submitWorkflow(
           new AbstractMap.SimpleEntry<>("docker", "debian:stable-slim"));
       try (OutputStream out = new FileOutputStream(tempOptionsFile.getFile())) {
         out.write(mapper.writeValueAsString(workflowOptions).getBytes(StandardCharsets.UTF_8));
+        logger.info(
+            "Wrote options to tmp file {} (Options omitted due to sensitive data",
+            tempOptionsFile.getFile().getPath());
       }
 
+      // Put the user email and workspace ID as labels on the workflow
+      // This is used in the UI.
       labels.put(WORKSPACE_ID_LABEL_KEY, workspaceId.toString());
       labels.put(USER_EMAIL_LABEL_KEY, userEmail);
+
+      // Copy the source wdl from GCS
+      var sourceWdlPath = tempWorkflowSourceFile.getFile().toPath();
       if (workflowGcsUri != null) {
         labels.put(GCS_SOURCE_LABEL_KEY, workflowGcsUri);
         InputStream inputStream =
             fileService.getFile(token, workspaceId, workflowGcsUri, /*convertTo=*/ null);
-        Files.copy(
-            inputStream,
-            tempWorkflowSourceFile.getFile().toPath(),
-            StandardCopyOption.REPLACE_EXISTING);
+        Files.copy(inputStream, sourceWdlPath, StandardCopyOption.REPLACE_EXISTING);
+        logger.info("Copied source WDL from {} to tmp file {}", workflowGcsUri, sourceWdlPath);
       }
       try (OutputStream out = new FileOutputStream(tempLabelsFile.getFile())) {
         out.write(mapper.writeValueAsString(labels).getBytes(StandardCharsets.UTF_8));
+        logger.info("Wrote labels {} to tmp file {}", labels, tempLabelsFile.getFile().getPath());
       }
 
-      // Get dependencies from GCS and create temp file
-      if (workflowDependenciesGcsUri != null) {
-        InputStream inputStream =
-            fileService.getFile(
-                token, workspaceId, workflowDependenciesGcsUri, /*convertTo=*/ null);
-        Files.copy(
-            inputStream,
-            tempWorkflowDependenciesFile.getFile().toPath(),
-            StandardCopyOption.REPLACE_EXISTING);
-      }
+      if (containsImportStatement(sourceWdlPath.toString())) {
```

Comment:
messed around with that but IMO the abstraction was weird because in the parseInputs call we do the zipping, which takes a bunch of arguments and it didn't really feel better imo